### PR TITLE
clickhouse_info: add limit argument

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
clickhouse_info: add limit argument

### CODE REVIEW

**Comment by Andersson007** [comment]
@aleksvagachev i don't know if it's helpful or not but I think limiting the output is a good idea considering the huge list of settings. Also users might not have all the permission for every system table like with system.users for example.
Feel free to approve and merge / raise concerns about the idea in general or implementation.

**Comment by aleksvagachev** [comment]
@Andersson007 I think this is a good idea. Shouldn’t we also add the ability to shoot everything, so as not to write each one separately, if someone needs this opportunity?

**Comment by Andersson007** [comment]
> @Andersson007 I think this is a good idea. Shouldn’t we also add the ability to shoot everything, so as not to write each one separately, if someone needs this opportunity?

@aleksvagachev could you give an example of a task?

**Comment by aleksvagachev** [comment]
@Andersson007 For the ansible.builtin.setup collection, you can select all. I thought it might be a good idea to add it here too. When testing, it is easier to specify everything at once. If the idea is bad, you can abandon it.

**Comment by Andersson007** [comment]
> @Andersson007 For the ansible.builtin.setup collection, you can select all. I thought it might be a good idea to add it here too. When testing, it is easier to specify everything at once. If the idea is bad, you can abandon it.

ah, i see now. When not specified, it gathers all, so not sure if we need an explicit value for that. Can you name a couple of scenarios when it'd be handy?

File: plugins/modules/clickhouse_info.py [hunk 1]
```diff
@@ -88,8 +96,17 @@
 - name: Print returned data
   ansible.builtin.debug:
     var: result
+
+- name: Limit return values with users and roles
+  register: result
+  community.clickhouse.clickhouse_info:
+    filter:
```
Comment (Andersson007) [inline plugins/modules/clickhouse_info.py:103]:
```suggestion
    limit:
```

**Comment by Andersson007** [comment]
> @Andersson007 I didn't notice that if nothing is specified, then everything is going to be collected. Maybe you should explicitly specify it in the documentation, for people like me)

Good point, will do, thanks! how about the argument name? Is it OK now or something like `limit_with` or `gather_subset` would be better?

**Comment by aleksvagachev** [comment]
All options are good, can we leave it as it is, as the shortest and most convenient?

**Comment by Andersson007** [comment]
@aleksvagachev updated the argument description to reflect your note, PTAL

**Review by aleksvagachev — APPROVED** [review]

**Comment by Andersson007** [comment]
@aleksvagachev thanks for reviewing and merging!

**Pull request merged**